assets/src/application/ui/input/EspressoButton/EspressoButton.tsx

```diff
@@ -14,15 +13,15 @@ type ButtonType = React.ComponentType<EspressoButtonProps>;
  * forwardRef to be able to accept
  * onMouseEnter, onMouseLeave, onFocus, onClick events from parent
  */
-const EspressoButton = React.forwardRef<Button, EspressoButtonProps>(
+const EspressoButton = React.forwardRef<typeof Button, EspressoButtonProps>(
 	(
 		{
-			icon,
-			onClick,
-			buttonText,
 			buttonSize = EspressoButtonSize.DEFAULT,
+			buttonText,
 			buttonType = EspressoButtonType.DEFAULT,
 			className: htmlClass,
+			icon,
```

Comment:
do we still need to pass this prop here, since we have that separated `EspressoIconButton`?
and also you proposed yesterday to have the separation of concerns in this component with which I do agree
also chakra have the same approach: https://chakra-ui.com/iconbutton

Reply:
We can pass the button that is shown on the left side of the text.

Reply:
`IconButton` cannot have text. So when we need both Icon and text, we will pass that icon and text to the button.